src/control/tools/StrokeHandler.cpp

```diff
@@ -369,9 +369,14 @@ void StrokeHandler::onButtonPressEvent(const PositionInputData& pos) {
         this->buttonDownPoint.x = pos.x / zoom;
         this->buttonDownPoint.y = pos.y / zoom;
 
-        createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y, pos.pressure));
+        this->hasPressure = this->xournal->getControl()->getToolHandler()->getToolType() == TOOL_PEN &&
+                            pos.pressure != Point::NO_PRESSURE;
 
-        this->hasPressure = this->stroke->getToolType() == STROKE_TOOL_PEN && pos.pressure != Point::NO_PRESSURE;
+        if (this->hasPressure) {
+            createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y, pos.pressure));
+        } else {
+            createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y));
+        }
```

Comment:
What's the difference to this?:
```suggestion
            createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y, pos.pressure));
```

Reply:
When the tool is the highlighter, then the event `pos` may have a pressure value, but we do not want to give the point that pressure value.
The boolean `this->hasPressure` is always `false` when the tool is the highlighter, and `true` when the tool is the normal pen and the device is pressure sensitve.

Reply:
Ok, looked into the source of the function.
Wouldn't it an idea to move that into it? Then the pressure is not set wrongly at other places?

Reply:
If not this LGTM and you can merge it ASAP.